pkg/acsengine/params_k8s.go

```diff
@@ -244,7 +254,17 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
 				CloudProviderRateLimitBucket: properties.OrchestratorProfile.KubernetesConfig.CloudProviderRateLimitBucket,
 			})
 			addValue(parametersMap, "kubeClusterCidr", properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet)
-			addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.OrchestratorProfile.KubernetesConfig.KubeletConfig["--non-masquerade-cidr"])
+			if properties.OrchestratorProfile.IsAzureCNI() {
+				if properties.MasterProfile != nil && properties.MasterProfile.IsCustomVNET() {
+					addValue(parametersMap, "kubernetesNonMasqueradeCidr", properties.MasterProfile.VnetCidr)
+				} else {
+					addValue(parametersMap, "kubernetesNonMasqueradeCidr", DefaultVNETCIDR)
+				}
+			} else if properties.HostedMasterProfile != nil {
```

Comment:
@JunSun17 Is it correct that we want a separate `kubernetesNonMasqueradeCidr` param value for scenarios where we have a `HostedMasterProfile`?